src/sentry/static/sentry/app/components/smartSearchBar.jsx

```diff
@@ -398,28 +478,63 @@ class SmartSearchBar extends React.Component {
         case 'firstSeen':
         case 'lastSeen':
         case 'event.timestamp':
-          out.className = 'icon-clock';
+          out.className = 'icon-av_timer';
           break;
         default:
           out.className = 'icon-tag';
       }
+
+      if (item.type === 'recent-search') {
+        out.className = 'icon-clock';
+      }
+
       return out;
     });
 
-    if (searchItems.length > 0 && !this.isDefaultDropdown()) {
+    if (searchItems.length > 0) {
       searchItems[0].active = true;
     }
 
     if (maxSearchItems && maxSearchItems > 0) {
       searchItems = searchItems.slice(0, maxSearchItems);
     }
 
+    const recentItems = recentSearchItems.map(item => ({
+      desc: item.query,
+      value: item.query,
+      className: 'icon-clock',
+      type: 'recent-search',
+    }));
+
     this.setState({
-      searchItems,
+      searchItems: [...searchItems, ...recentItems],
       activeSearchItem: 0,
     });
   };
 
+  onTogglePinnedSearch = evt => {
+    const {
+      api,
+      organization,
+      savedSearchType,
+      hasPinnedSearch,
+      pinnedSearch,
+    } = this.props;
+
+    evt.preventDefault();
+    evt.stopPropagation();
+
+    if (!defined(savedSearchType) || !hasPinnedSearch) {
+      return;
+    }
+
+    if (!!pinnedSearch) {
+      unpinSearch(api, organization.slug, savedSearchType);
+    } else {
+      pinSearch(api, organization.slug, this.state.query, savedSearchType);
```

Comment:
small nit: could be nice to switch the order of the last 2 variables so `unpin` and `pin` have a similar interface

Reply:
Done